integrations/operator/controllers/resources/github_connector_controller.go

```diff
@@ -38,6 +38,8 @@ func (r githubConnectorClient) Get(ctx context.Context, name string) (types.Gith
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	teleportClient.RLock()
```

Comment:
Can't we hide this by wrapping `teleportClient` with something that removes the requirement for having locks per call? 
``` 
func (t *teleportClient) GetGithubConnector(ctx, name, false /* with secrets*/){
t.Rlock()
defer t.RUnlock()
return t.client.GetGithubConnector(ctx, name, false /* with secrets*/)

}
```

Reply:
Yes, but the `githubConnectorClient` is already a wrapper around the teleport client. If we add wrapper methods that handle locking on the `sidecar.SyncClient`, we'll need to write one for each call (the only place where we can gain is when create/update are both backed by upsert)

Adding a new resource would imply adding the same amount of locks, we don't really reduce the probability of shooting ourselves in the foot.

Reply:
I would still prefer some sort sync.WaitGroup + retiring channel where wg.Done is called when closing teleportClient but looks exactly the same approach.